src/Parser.php

```diff
@@ -2559,39 +2559,47 @@ private function parseNamespaceUseDeclaration($parentNode) {
 
         $namespaceUseDeclaration->useKeyword = $this->eat(TokenKind::UseKeyword);
         $namespaceUseDeclaration->functionOrConst = $this->eatOptional(TokenKind::FunctionKeyword, TokenKind::ConstKeyword);
-        $namespaceUseDeclaration->namespaceName = $this->parseQualifiedName($namespaceUseDeclaration);
-        if (!$this->checkToken(TokenKind::OpenBraceToken)) {
+        if($this->lookahead(TokenKind::CommaToken)) {
```

Comment:
Yea, lookahead for comma did seem like a poor design. So with these node revisions, we should end up with `NamespaceUseDeclaration` having `useClauses`, that always assumes theres a list (`NamespaceUseClauseList`), even with just one member? If so, whats the purpose of `groupClauses`  inside the new `NamespaceUseDeclaration`?